### ISSUE TITLE
kubernetes cluster does not have "default storage class". 

### ISSUE THREAD

**Issue by bayeslearner** [karmab/kcli#581]
I'm trying to install kubesphere on top of it, but it complains about not finding a default storageClass. 
```
kubectl get storageclass
No resources found
```

```
TASK [preinstall : KubeSphere | Stopping if default StorageClass was not found] ***
fatal: [localhost]: FAILED! => {
    "assertion": "\"(default)\" in default_storage_class_check.stdout",
    "changed": false,
    "evaluated_to": false,
    "msg": "Default StorageClass was not found !"    }

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=2    unreachable=0    failed=1    skipped=4    rescued=0    ignored=0
``

**Comment by bayeslearner** [comment]
so I see that nfs server is running on the ctrplane node.  
I thought by "nfs: true" as the default for generic kube,  you meant nfs storage class will be  configured during cluster installation.  
I guess I misunderstand?

**Comment by karmab** [comment]
the default nfs boolean only creates 20 static pvs using storage from the first ctlplane node.

I've added support for using nfs subprovisioner with a new variable named `nfs_sc` (which defaults to False) in https://github.com/karmab/kcli/commit/d94eedab514a22222909fae97fcfeee21ba79f83. That will give you a default sc/dynamic provisioning.

Note that it's not an ideal solution, cos we still have a SPOF from the first control plane node. We could improve the situation by pointing to some remote nfs share but I decided to keep the code simple for now
